mitodl/micromasters
# Added home page signup popup

#### What are the relevant tickets?

closes #917 

#### What's this PR do?

This adds a little dialog to the home page and to the signup pages which lets you pick a program. We persist this to `localStorage`. In the dashboard, when the `App` container updates it checks for this value in `localStorage` and, if present, it uses it to issue an `addProgramEnrollment` api request.

#### Where should the reviewer start?

#### How should this be manually tested?

right now:

- go to the homepage
- run `window.openDialog()` in the JS console
- pick a program your user is not currently enrolled in
- go to the dashboard
- confirm that, after the dashboard loads, `localStorage.programId` is empty, and that your user has a new program enrollment.

#### Any background context you want to provide?

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
